Assembus/Assets/Scripts/MainScreen/Sidebar/HierarchyView/HierarchyViewController.cs

```diff
@@ -469,5 +492,61 @@ private void DeselectParent(Transform child)
             if (SelectedItems.Contains(parent)) DeselectItem(parent);
             else DeselectParent(parent.transform);
         }
+
+        /// <summary>
+        ///     Scrolls to the targetItem
+        /// </summary>
+        /// <param name="targetItem"></param>
+        public void ScrollToItem(RectTransform targetItem)
+        {
+            scroll.enabled = false;
+            Canvas.ForceUpdateCanvases();
+            scroll.enabled = true;
+
+            // Top border of the actual viewport
+            var topBorder = contentPanel.localPosition.y;
+
+            // Lower border of the actual viewport
+            var lowerBorder = topBorder + 1000;
+
+            // Target item position in the viewport
+            var itemPosition = scrollRect.InverseTransformPoint(contentPanel.position).y -
+                               scrollRect.transform.InverseTransformPoint(targetItem.position).y;
+
+            // Check if item is outside the borders, if so, scroll to the item
+            if (!(itemPosition < lowerBorder && topBorder < itemPosition))
+                StartCoroutine(ScrollToTarget(itemPosition - 200));
+        }
+
+        /// <summary>
+        ///     Coroutine for smoth scrolling to an new item
+        /// </summary>
+        /// <param name="targetValue"></param>
+        /// <returns></returns>
+        private IEnumerator ScrollToTarget(float targetValue)
+        {
+            var interpolatedFloat = new InterpolatedFloat(contentPanel.anchoredPosition.y);
+
+            while (interpolatedFloat != targetValue)
```

Comment:
Have you tried to fix the warning in this line? I know it technically is wrong but maybe it can be fixed?